app/src/main/java/com/davidtomas/taskyapp/features/auth/domain/useCase/ValidateLoginInputsUseCase.kt

```diff
@@ -0,0 +1,50 @@
+package com.davidtomas.taskyapp.features.auth.domain.useCase
+
+import com.davidtomas.taskyapp.core.domain.util.Result
+import com.davidtomas.taskyapp.features.auth.domain.model.AllFieldsLoginValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.EmailLoginValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.LoginInputValidationError
+import com.davidtomas.taskyapp.features.auth.domain.model.LoginInputValidationErrors
+import com.davidtomas.taskyapp.features.auth.domain.model.LoginValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.PasswordLoginValidationType
+
+class ValidateLoginInputsUseCase(
+    private val validateEmailUseCase: ValidateEmailUseCase,
+    private val validatePasswordUseCase: ValidatePasswordUseCase
+) {
+
+    private val loginInputValidationError = mutableSetOf<LoginInputValidationError>()
+
+    operator fun invoke(
+        loginValidationType: LoginValidationType
+    ): Result<Unit> {
+        loginInputValidationError.clear()
+        when (loginValidationType) {
+            is EmailLoginValidationType -> {
+                validateEmailUseCase.invoke(loginValidationType.email)?.let {
```

Comment:
What's the point of overriding the invoke operator and then still calling the function by its name? :D